server/repository/src/db_diesel/program_requisition/program_supplier.rs

```diff
@@ -0,0 +1,314 @@
+        // should result in nothing (since name1 is not store)
+        let repo = ProgramSupplierRepository::new(&connection);
+
+        let filter = ProgramSupplierFilter::new()
+            .program_id(EqualFilter::equal_any(vec![
+                program1.id.clone(),
+                program2.id.clone(),
+            ]))
+            .name(NameFilter::new().is_store(true).is_visible(true));
+
+        let result = repo.query(&store3.id, filter);
+
+        assert_eq!(result, Ok(Vec::new()));
+
+        // TEST 2 with master list join for store 1 and without name_store_join for store 2
+        // should result in one found, store_name1
+        let filter = ProgramSupplierFilter::new()
+            .program_id(EqualFilter::equal_any(vec![
+                program1.id.clone(),
+                program2.id.clone(),
+            ]))
+            .name(NameFilter::new().is_store(true).is_visible(true));
+
+        MasterListNameJoinRepository::new(&connection)
+            .upsert_one(&master_list_name_join2)
+            .unwrap();
+
+        let result = repo.query(&store3.id, filter);
+
+        assert_eq!(
+            result,
+            Ok(vec![ProgramSupplier {
+                supplier: Name {
+                    name_row: store_name1.clone(),
+                    name_store_join_row: Some(name_store_join2.clone()),
+                    store_row: Some(store1.clone()),
+                },
+                program: program1.clone()
+            }])
+        );
+
+        // TEST 3 with master list join for store 1 and with name_store_join for store 2
+        // should result in two found, store_name1 and store_name2
+        let filter = ProgramSupplierFilter::new()
+            .program_id(EqualFilter::equal_any(vec![
+                program1.id.clone(),
+                program2.id.clone(),
+            ]))
+            .name(NameFilter::new().is_store(true).is_visible(true));
+
+        NameStoreJoinRepository::new(&connection)
+            .upsert_one(&name_store_join3)
+            .unwrap();
+
+        let result = repo.query(&store3.id, filter);
+
+        assert_eq!(
+            result,
+            Ok(vec![
+                ProgramSupplier {
+                    supplier: Name {
+                        name_row: store_name1.clone(),
+                        name_store_join_row: Some(name_store_join2.clone()),
+                        store_row: Some(store1.clone()),
+                    },
+                    program: program1.clone()
+                },
+                ProgramSupplier {
+                    supplier: Name {
+                        name_row: store_name2.clone(),
+                        name_store_join_row: Some(name_store_join3.clone()),
+                        store_row: Some(store2.clone()),
+                    },
+                    program: program2.clone()
+                }
+            ])
+        );
+
+        //
```

Comment:
:nit not needed?

Reply:
goneski: 05eedbefa935b9985dc682308f07256797ee77dd